src/test/java/attendance/AttendanceBookTest.java

```diff
@@ -0,0 +1,234 @@
+        ),
+        Arguments.arguments(
+            LocalDateTime.of(2025, 2, 27, 10, 6),
+            AttendanceStatus.LATE
+        ),
+        Arguments.arguments(
+            LocalDateTime.of(2025, 2, 28, 10, 31),
+            AttendanceStatus.ABSENCE
+        )
+    );
+  }
+
+  @DisplayName("출석_수정_시_등록되지_않은_크루_예외")
+  @Test
+  void validateExistentCrewTest() {
+    assertThatThrownBy(
+        () -> attendanceBook.modify("없는크루", LocalDateTime.of(2025, 2, 28, 10, 0))
+    ).isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("등록되지 않은 크루입니다");
+  }
+
+  @DisplayName("출석_수정_시_수정_전_기록과_수정_후_기록_반환")
+  @Test
+  void modifyRecordReturnTest() {
+    ExistentAttendanceRecord existentAttendanceRecord = attendanceBook.check("히포",
+        LocalDateTime.of(2025, 2, 28, 10, 12));
+
+    LocalDateTime updateDateTime = LocalDateTime.of(2025, 2, 28, 10, 2);
+    Entry<AttendanceRecord, AttendanceRecord> updatedAttendanceRecords = attendanceBook.modify(
+        "히포", updateDateTime);
+
+    assertThat(updatedAttendanceRecords.getKey().getRecord()).isEqualTo(
+        existentAttendanceRecord.getRecord());
+    assertThat(updatedAttendanceRecords.getValue().getRecord()).isEqualTo(updateDateTime);
+  }
+
+  @DisplayName("출석_기록이_없는_경우에_대한_수정_전_기록과_수정_후_기록_반환")
+  @Test
+  void modifyAbsenceRecordReturnTest() {
+    LocalDateTime updateDateTime = LocalDateTime.of(2025, 2, 28, 10, 2);
+    Entry<AttendanceRecord, AttendanceRecord> updatedAttendanceRecords = attendanceBook.modify(
+        "히포", updateDateTime);
+
+    assertThat(updatedAttendanceRecords.getKey().getRecord()).isEqualTo(
+        LocalDateTime.of(2025, 2, 28, 0, 0));
+    assertThat(updatedAttendanceRecords.getValue().getRecord()).isEqualTo(updateDateTime);
+  }
+
+  @DisplayName("출석_불가능한_날_수정_시도_시_예외_발생")
+  @Test
+  void modifyRecordToUnavailableDateTest() {
+    LocalDateTime updateDateTime = LocalDateTime.of(2025, 3, 1, 10, 2);
+    assertThatThrownBy(
+        () -> attendanceBook.modify("히포", updateDateTime)
+    ).isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("주말 또는 공휴일에는 운영하지 않습니다.");
+  }
+
+  @DisplayName("출석_불가능한_시간_수정_시도_시_예외_발생")
+  @Test
+  void modifyRecordToUnavailableTimeTest() {
+    LocalDateTime updateDateTime = LocalDateTime.of(2025, 2, 28, 7, 2);
+    assertThatThrownBy(
+        () -> attendanceBook.modify("히포", updateDateTime)
+    ).isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("캠퍼스 운영 시간이 아닙니다");
+  }
+
+  @DisplayName("등록되지_않은_크루_출석_조회_시_예외_발생")
+  @Test
+  void notExistsCrewAttendanceRecordsSearchingTest() {
+    assertThatThrownBy(() ->
+        attendanceBook.search("없는크루", LocalDate.of(2025, 2, 11))
+    ).isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("등록되지 않은 크루입니다");
+  }
+
+  @DisplayName("출석_기록과_상태를_정렬하여_조회")
+  @Test
+  void searchedRecordTest() {
+    setRecords();
+
+    LinkedHashMap<LocalDateTime, AttendanceStatus> searched = attendanceBook.search("히포",
```

Comment:
```suggestion
    Map<LocalDateTime, AttendanceStatus> searched = attendanceBook.search("히포",
```
추상화라는 개념에 대해서 조금 더 의식해보세요. 추상화를 하는 이유가 무엇일까요?

Reply:
추상화를 사용하는 이유 중 유연성과 확장성에 대한 고민이 부족한 코드였네요..!

메서드 반환 타입을 구체 타입으로 지정하면서 실제로 반환하는 값이 변경되는 상황에 대한 유연한 대처가 불가능해졌다는 생각이 들었습니다 😂